### PR TITLE
cargohold: add compatibility mode for Scality RING S3 implementation

### PR DESCRIPTION
**Pull request by mheinzel**
See https://github.com/zinfra/backend-issues/issues/1659

Not the nicest way to do it and doesn't adhere to the S3 spec, but at least this way it's behind an optional setting.

### CODE REVIEW

**Review by jschaul — APPROVED** [review]
I can confirm this works against the different compatibility layers (aws, minio, scality-ring).

File: services/cargohold/src/CargoHold/Options.hs [hunk 1]
```diff
@@ -51,11 +52,24 @@ data AWSOpts = AWSOpts
     _awsS3DownloadEndpoint :: !(Maybe AWSEndpoint),
     -- | S3 bucket name
     _awsS3Bucket :: !Text,
+    -- | Enable this option for compatibility with specific S3 backends.
+    _awsS3Compatibility :: !(Maybe (Last S3Compatibility)),
```
Comment (jschaul) [inline services/cargohold/src/CargoHold/Options.hs:56]: Just wondering: Why wrap this in `Last` ?

**Comment by jschaul** [comment]
(We also need to expose this optional config flag in wire-server-deploy)

File: services/cargohold/src/CargoHold/Options.hs (hunk 1 above)
Comment (mheinzel) [reply services/cargohold/src/CargoHold/Options.hs]: Ah, yeah, I saw it used in other options and was wondering about it as well. Forgot to remove it, but it's done now and apparently didn't have any deeper reason.

**Comment by mheinzel** [comment]
> (We also need to expose this optional config flag in wire-server-deploy)

https://github.com/wireapp/wire-server-deploy/pull/364

**Pull request merged**